File: src/test/java/CombatIntegrationTest.java
import org.junit.Test;

import java.util.Random;
import static org.junit.Assert.*;

/**
 * Created by andrew on 7/1/16.
 */
public class CombatIntegrationTest {

    @Test
    public void basicCombatTest() {
        CharacterFactory cf = new CharacterFactory();

        Character human1 = cf.buildBasicHuman();
        Character human2 = cf.buildBasicHuman();

        Random r = new Random();
        r.setSeed(42);
        Combat c = new Combat(r);

        assertEquals(180, human2.getHealth());
        boolean hit = c.hit(human1, human1.getPrimaryWeapon());
        int damage = 0;
        if (hit) {
            damage = c.applyAttackDamage(human1, human1.getPrimaryWeapon(), human2);
        }
        assertTrue(hit);
        assertEquals(15, damage);
        assertEquals(165, human2.getHealth());
    }
}

File: src/test/java/CombatIntegrationTest.kt
import org.junit.Test

import java.util.Random
import org.junit.Assert.*

class CombatIntegrationTest {

    @Test
    fun basicCombatTest() {
        val cf = CharacterFactory()

        val human1 = cf.buildBasicHuman()
        val human2 = cf.buildBasicHuman()

        val r = Random()
        r.setSeed(42)
        val c = Combat(r)

        assertEquals(180, human2.health.toLong())
        val hit = c.hit(human1, human1.primaryWeapon!!)
        var damage = 0
        if (hit) {
            damage = c.applyAttackDamage(human1, human1.primaryWeapon!!, human2)
        }
        assertTrue(hit)
        assertEquals(15, damage.toLong())
        assertEquals(165, human2.health.toLong())
    }
}
